src/setup_payload/java/src/chip/setuppayload/SetupPayloadParser.java

```diff
@@ -4,14 +4,37 @@
 public class SetupPayloadParser {
 
   /** Returns {@link SetupPayload} parsed from the QR code string. */
-  public SetupPayload parseQrCode(String qrCodeString) throws UnrecognizedQrCodeException {
-    return fetchPayloadFromQrCode(qrCodeString);
+  public SetupPayload parseQrCode(String qrCodeString)
+      throws UnrecognizedQrCodeException, SetupPayloadException {
+    return fetchPayloadFromQrCode(qrCodeString, false);
+  }
+
+  /**
+   * Returns {@link SetupPayload} parsed from the QR code string.
+   *
+   * @param qrCodeString the QRCode for commissioning device.
+   * @param allowInvalidPayload Allow invalid payload components.
```

Comment:
Please provide examples. It is obvious that 'allowInvalidPayload' 'allows invalid payloads'.
The documentation should explain non-obvious things. like what is an invalid payload? why would one allow it? What is the return value if the payload is invalid but allowed? are any payloads allowed when this is true?

Reply:
Added additional description.
This value is added for the same purpose as chiptool's "--allow-invalid-payload" option. In the case of the Android platform, since commissioning is performed after parsing the pairing code, both cases are considered necessary. (I think it is necessary for the user to only parse the QR Code / Manual Pairing Code.)